### PR TITLE
Tolerate execeptions for periodical scheduled tasks

### PR DESCRIPTION
**Pull request by bowenliang123**
# :mag: Description
## Issue References 🔗

As the PR https://github.com/apache/kyuubi/pull/5727 fixed a problem caused by leaking exception in a scheduled task, the exception thrown should be properly handled to prevent suspension in `ScheduledExecutorService.scheduleWithFixedDelay`.

## Describe Your Solution 🔧
Introducing a util method `scheduleTolerableRunnableWithFixedDelay` for catching possible exceptions in scheduled tasks.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Behavior Without This Pull Request :coffin:
No behaviour change.

#### Behavior With This Pull Request :tada:
This is a fallback measure to ensure all the exceptions handled. 
Currently all the occurrences are properly handled.

#### Related Unit Tests
Pass all the CI tests.

---

# Checklists
## 📝 Author Self Checklist

- [x] My code follows the [style guidelines](https://kyuubi.readthedocs.io/en/master/contributing/code/style.html) of this project
- [x] I have performed a self-review
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

## 📝 Committer Pre-Merge Checklist

- [ ] Pull request title is okay.
- [ ] No license issues.
- [ ] Milestone correctly set?
- [ ] Test coverage is ok
- [ ] Assignees are selected.
- [ ] Minimum number of approvals
- [ ] No changes are requested

**Be nice. Be informative.**

### CODE REVIEW

**Comment by bowenliang123** [comment]
cc @turboFei @wForget

File: kyuubi-common/src/main/scala/org/apache/kyuubi/util/ThreadUtils.scala [hunk 1]
```diff
@@ -109,4 +109,28 @@ object ThreadUtils extends Logging {
...
+      delay: Long,
+      timeUnit: TimeUnit,
+      errorMessageHint: String): Unit = {
+    scheduler.scheduleWithFixedDelay(
+      () =>
+        try {
+          runnable.run()
+        } catch {
+          case e: Exception =>
+            error(s"Error in $errorMessageHint", e)
```
Comment (turboFei) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/util/ThreadUtils.scala:130]:
```
case e: Throwable => error(s"Uncaught exception in thread ${Thread.currentThread().getName}", t)
```

Comment (turboFei) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/util/ThreadUtils.scala]: You can remove `errorMessageHint` variable.

It is ok to use the thread name as hint.

Comment (turboFei) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/util/ThreadUtils.scala]:
```
case e: Throwable => error(s"Uncaught exception in thread ${Thread.currentThread().getName}", t)
```

Comment (bowenliang123) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/util/ThreadUtils.scala]: Sure, it's much more elegant.

**Review by turboFei — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merged to master and branch-1.8

**Pull request closed without merging**